flumine/order/process.py

```diff
@@ -59,10 +59,8 @@ def process_current_order(order: BaseOrder):
         elif order.size_remaining == 0:
             order.execution_complete()
 
-    if order.status == OrderStatus.EXECUTION_COMPLETE:
-        if order.trade.status == TradeStatus.LIVE:
-            if not order.trade.offset_orders:
-                order.trade.complete()
+    if order.trade.status == TradeStatus.LIVE and order.trade.trade_complete:
```

Comment:
Bug here where it will complete the trade and reset the strategy when between the cancel->place in execution.replace

Can't see an easy fix here without modify the TradeStatus when carrying out an execution, maybe a clever context manager in execution?

```python
Trade.__enter__:
    self.update_status(TradeStatus.PENDING)

Trade.__exit__:
    self.update_status(TradeStatus.LIVE)
```

And then in execution:

```python
with order.trade:
    # process cancel response
    if instruction_report.cancel_instruction_reports.status == "SUCCESS":
        ..
```